From c13bfaa8ca023923cc42d0808fc4f4f83260b2f7 Mon Sep 17 00:00:00 2001
From: elupus <elupus@svn>
Date: Thu, 15 Jul 2010 22:24:57 +0000
Subject: [PATCH] fixed: dvd still frames as first frame caused lavf to return CODEC_ID_PROBE, aswell as put the whole menu into an internal buffer.

This fixes #9470 aswell as makes our (incomplete) hacks in DVDDemuxerFFMpeg for this case redundant. Getting this upstream would be good, but we'd need a mpeg sample file. One can be generated by chopping a .vob file at a location where there is only a singular still frame available.

git-svn-id: https://xbmc.svn.sourceforge.net/svnroot/xbmc/trunk@31831 568bbfeb-2a22-0410-94d2-cc84cf5bfa90
---
 .../dvdplayer/Codecs/ffmpeg/libavformat/utils.c    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git ag/libavformat/utils.c b/xbmc/cores/dvdplayer/Codecs/ffmpeg/libavformat/utils.c
index 2e623e2..7bc89f8 100644
--- ag/libavformat/utils.c
+++ bg/libavformat/utils.c
@@ -702,7 +702,7 @@ int av_read_packet(AVFormatContext *s, AVPacket *pkt)
 
             if(av_log2(pd->buf_size) != av_log2(pd->buf_size - pkt->size)){
                 //FIXME we dont reduce score to 0 for the case of running out of buffer space in bytes
-                set_codec_from_probe_data(s, st, pd, st->probe_packets > 0 ? AVPROBE_SCORE_MAX/4 : 0);
+                set_codec_from_probe_data(s, st, pd, st->probe_packets > 0 ? AVPROBE_SCORE_MAX/4-1 : 0);
                 if(st->codec->codec_id != CODEC_ID_PROBE){
                     pd->buf_size=0;
                     av_freep(&pd->buf);
-- 
1.7.3

